crates/sui-core/src/authority.rs

```diff
@@ -860,6 +878,30 @@ impl AuthorityState {
                 Ok(res) => res,
             };
 
+        // Write tx output to WAL as first commit phase. In second phase
+        // we write from WAL to permanent storage. The purpose of this scheme
+        // is to allow for retrying phase 2 from phase 1 in the case where we
+        // fail mid-write. We prefer this over making the write to permanent
+        // storage atomic as this allows for sharding storage across nodes, which
+        // would be more difficult in the alternative.
+        self.database.wal.write_execution_output(
+            &digest,
+            inner_temporary_store.clone(),
+            signed_effects.clone(),
+        )?;
+
+        self.commit_cert_and_notify(certificate, inner_temporary_store, signed_effects, tx_guard)
+            .await
+    }
+
+    async fn commit_cert_and_notify(
+        &self,
+        certificate: &VerifiedCertificate,
+        inner_temporary_store: InnerTemporaryStore,
+        signed_effects: SignedTransactionEffects,
+        tx_guard: CertTxGuard<'_>,
+    ) -> SuiResult<VerifiedTransactionInfoResponse> {
+        let digest = *certificate.digest();
```

Comment:
Are we also somewhere cleaning up the transaction entry from the wal once it is committed? Given we are going to store tx effects there, it might consume significant disk space.

Reply:
When we call `commit_tx` or `release()` on the TxGuard, it deletes the wal entry for that tx.